ros_ethercat_model/include/ros_ethercat_model/ros_ethercat.hpp

```diff
@@ -182,6 +182,54 @@ class RosEthercat : public hardware_interface::RobotHW
     }
   }
 
+  /// Update priority of provided thread
+  bool updateThreadPriority(boost::thread& a_thread)
+  {
+    int retcode;
+    int policy;
+    struct sched_param param;
+
+    pthread_t threadID = (pthread_t) a_thread.native_handle();
+
+    policy = SCHED_FIFO;
+    param.sched_priority = sched_get_priority_max(policy);
+
+    if ((retcode = pthread_setschedparam(threadID, policy, &param)) != 0)
```

Comment:
do we need the `retcode` variable?

Reply:
We don't, good spot https://github.com/shadow-robot/ros_ethercat/pull/122/commits/81a7232103c455f1cc2aaf3b5a6774daa57d0d2a